holoviews/util/__init__.py

```diff
@@ -244,6 +246,30 @@ def defaults(cls, *options, **kwargs):
         cls._linemagic(cls._expand_options(merge_options_to_dict(options)), backend=kwargs.get('backend'))
 
 
+    @classmethod
+    def _expand_by_backend(cls, options, backend):
+        "Given a list of Option objects, return a list of grouped backend"
+        groups = defaultdict(list)
+        used_fallback = False
+        for obj in options:
+            if obj is None: # Why are None's being supplied?
+                continue
+            elif obj.backend:
+                opts_backend = obj.backend
+            elif backend is None:
+                opts_backend = Store.current_backend
+                obj.backend= opts_backend
+            else:
+                opts_backend = backend
+                obj.backend = opts_backend
+                used_fallback = True
+            groups[opts_backend].append(obj)
+
+        if backend and not used_fallback:
+            cls.param.warning('Fallback backend specified but not used.')
```

Comment:
This warning should be more informative, it's not clear what a fallback backend is. 

```suggestion
            cls.param.warning("All supplied Options objects already define a backend, "
                              "backend override backend=%r will be ignored." % backend)
```

Reply:
Thanks for the suggestion. What I wrote was a WIP placeholder...